### PR TITLE
Start porting ergo-tree constants

### PR DESCRIPTION
**Pull request by ross-weir**
@greenhat 

Will add the rest of the `Constant` methods in next PRs, just want to check that you're happy with the direction I'm going.

Allowing more flexibility in how `Constant`s are created as opposed to specific constructor methods like `Constant.from_coll_coll_byte` like we have at the moment. `constants.test.ts` tests should show examples of before vs after

### CODE REVIEW

File: packages/ergo-lib-wasm/tests/ir/constants.test.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { SColl, SByte, SInt } from "../../";
+
+describe("Constants", () => {
+  describe("Functional parity with library v0.x.x", () => {
+    describe("Constant.from_byte_array(new Uint8Array([3, 2, 1]))", () => {
+      it("should serialize to the same value", () => {
+        const constant = new SColl([
+          new SByte(3),
```
Comment (greenhat) [inline packages/ergo-lib-wasm/tests/ir/constants.test.ts:8]: Looks nice!
Do you think we need to expose `SType` into JS? I mean, there are `Constant.to_js` and `from_js` that should convert any type Rust<->JS (well, tuples are ugly).

Comment (ross-weir) [reply packages/ergo-lib-wasm/tests/ir/constants.test.ts:8]: > Looks nice! Do you think we need to expose `SType` into JS? I mean, there are `Constant.to_js` and `from_js` that should convert any type Rust<->JS (well, tuples are ugly).

By `SType` do you mean the `SByte` class for example?

I didn't really like the look of `Constant.from_js` & `Constant.to_js` because it only supports a subset of constants and looks ambiguous/brittle. For example, there's no way to create a `Coll` of `Short` because JS numbers are always converted to `Int` and we only support passing `Uint8Array` to get a `Coll` of `Byte`. Also we have situations like, if I pass a Uint8Array to `Constant.from_js` there's a number of `Constant`s that can be created from byte arrays, `SColl[Byte]`, `GroupElement`, `AvlTree`, etc

tl;dr I think it's worth exposing the types so devs can construct exactly what they need without ambiguity - We could add convenience methods at the `SType` level like `SColl.fromUint8Array`, etc

Comment (greenhat) [reply packages/ergo-lib-wasm/tests/ir/constants.test.ts:8]: Oh, you're right, I totally forgot about that (precise types). Good call!

Comment (ross-weir) [reply packages/ergo-lib-wasm/tests/ir/constants.test.ts:8]: > Oh, you're right, I totally forgot about that (precise types). Good call!

Cool! Sounds like you're OK with this direction so I'll fix-up `Long` in this PR to use `BigInt` to avoid the mantissa limit then continue on with the other constant types in a follow up PR unless you have any objections :)

Comment (greenhat) [reply packages/ergo-lib-wasm/tests/ir/constants.test.ts:8]: Sounds great! Please do! I'm totally onboard!
